### ISSUE TITLE
fix: App hangs in black screen when adding a leading to a ListView on iOS

### ISSUE THREAD

**Issue by erickzanardo** [shorebirdtech/shorebird#2237]
App ID: f71308f8-40fc-4db6-8403-e3dfc33683d8

**Description**

While trying to make a smaller app for scroll issue, I ended up finding another one.

When in a patch, I add a `leading` to a `ListTile` in a `ListView`, the app will forever hang in a black screen, there are not logs or errors on the console.

**Steps To Reproduce**

1. Create a new flutter app for iOS
2. Replace `main.dart` with:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
        body: ListView.builder(
          itemCount: 100,
          itemBuilder: (context, idx) {
            return ListTile(
              title: Text('Item $idx'),
            );
          },
        ),
      ),
    ),
  );
}
```

3. Configure shorebird and iOS certs and make a release
4.  Add the following `leading` bellow the `title` to the `ListTile`:
```diff
            return ListTile(
              title: Text('Item $idx'),
+             leading: Text('A'),
            );
```
5. Generate patch
6. When the app is launched with the new patch, it will hang in a black screen.

**Expected Behavior**

App renders the ListView.

**Additional Context**

Debug files:
[patch-debug.zip](https://github.com/user-attachments/files/15829090/patch-debug.zip)

**Comment by eseidel** [comment]
Thank you for the repro!

**Comment by hoyaaaa** [comment]
APP_ID: 0f2ace62-acfa-455a-b68e-9c644bfce83c
PATCH_NO: #2

I removed a component from the existing code and pushed it without making any other modifications or additions, but the screen remains black and does not progress (only on iOS).
So I have deactivated PATCH#2 for now.
After that, I tried to push again, but the code pushes are not progressing; they just hang with only the time taken increasing.

**Comment by eseidel** [comment]
See also https://github.com/shorebirdtech/shorebird/issues/2120 and https://github.com/shorebirdtech/shorebird/issues/1753 and https://github.com/shorebirdtech/shorebird/issues/1140

**Comment by eseidel** [comment]
Felix figured out today that this was a regression from our global dispatch table offset alignment fixes last week.  We're not entirely sure why, but reverting the change does resolve the issue, so we'll be releasing a revert as soon as tomorrow.

**Comment by eseidel** [comment]
We believe this should now be resolved for new releases made with Flutter 3.22.2 after today.

**Comment by eseidel** [comment]
I don't think this was actually fixed, but we fixed it in https://github.com/shorebirdtech/shorebird/issues/2219 just today.
